Tweaked D2k warhead falloffs

reaperrr opened the pull request:
I recently played the original quite a bit and I think our area of effect on several weapons is over the top even compared to the original, which is one of the reasons why debris are too powerful on bleed (they are being dealt with in #9615).

I tried to keep things as simple and consistent as possible, but this might need more testing and fine-tuning.

reaperrr committed:
Tweaked D2k warhead falloffs

pchote commented on mods/d2k/weapons.yaml:
```diff
@@ -5,8 +5,8 @@ LMG:
-		Spread: 48
-		Falloff: 100, 100, 65, 30, 0
+		Spread: 96
+		Falloff: 100, 50, 25, 12, 6, 0
```
Is there a specific reason why you didn't go back to our default falloff model for these? (the two are basically equal if you scale the spread by a factor of ~1.4 and don't mind the wider negligible-damage tail).

reaperrr committed:
Remove redundant 100% versus values from D2k warheads

reaperrr committed:
Make D2k bazooka and quad rocket more inaccurate

reaperrr committed:
Remove unnecessary D2k ValidTargets entries

reaperrr commented:
Updated.

- Using default Falloff in favor of larger Spread values
- increased machinegun spread a bit more to account for infantry's smaller health radius
- added some cleanup commits

pchote commented:
I spectated an all bot match and didn't spot anything bogus, so :+1:

penev92 commented:
Damage spreads are ridiculously high.
Quad rockets deal damage in a 3-cell diameter. Trike machine gun deals damage in a 4 (!) cell diameter.
Infantry's rifles deal damage in a 2-cell diameter and bazookas - in 3.

This both looks ridiculous when you enable Combat Geometry in the debug menu and causes infantry to go prone while not even being fired at.

reaperrr commented:
I guess this needs a custom Falloff afterall.

reaperrr commented:
Added back custom falloffs.

- for machineguns, I left Spread untouched and only added a custom Falloff of `100, 40, 16, 0`, which means damage is slightly higher at first and second spread, but then falls off a cliff compared to internal default to avoid adjacent infantry going prone.
- for anti-tank weapons, I also left Spread untouched and added a custom Falloff of `100, 50, 25, 0`, which means damage is moderately higher at first two spread ranges but then falls off a cliff, too.
- for high-explosive weapons, I slightly reduced Spread a bit in some cases (siege tank 155mm from 448 to 384) and added a custom Falloff of `100, 60, 30, 15, 0`, so the total area of effect is still significantly lower than with default Falloff + previous Spread, but damage falloff *within* area of effect is not as steep, to better reflect original behavior.

Don't merge this yet, there are some minor things I noticed which I want to adress first.

reaperrr committed:
Fix animations and sounds of explosions

Animation for small and medium explosion was reversed, sounds were wrong compared to original.

reaperrr committed:
Remove SpiceExplosion impact sound

Way too annoying, and the original didn't have this either.

reaperrr commented:
Updated.

- fine-tuned machine gun falloffs a bit more
- swapped explosion animation of small and medium unit explosion to match original
- fixed unit explosion sounds to match original as well
- removed spice explosion impact sound, was way too annoying

penev92 commented:
I noticed two things, the first one of which is unrelated to this PR:
 - Combat tanks are surprisingly inaccurate;
 - The Siege tank, while having a good spread, perhaps has too much of a Falloff, because the spread damage is negledgeable.

Other than that, the spreads look a lot more sane now.

reaperrr committed:
Re-increase siege tank and grenadier area of effect

Siege Tank:
- 'Stretches' each Spread step by 1 pixel (from 12 to 13), increasing total area of effect from 48 pixels (1.5 cells) to 52 pixels (1.625 cells)
- increases falloff damage by 5 percentage points

Grenadier:
- same Falloff as Siege Tank

reaperrr commented:
Updated.

- Reduced Siege Tank and Grenadier `Falloff` by 5 points at stages 2-4.
- Slightly re-increased Siege Tank `Spread` from 384 (12 pixel) to 416 (13 pixel), to stretch out falloff slightly more.

penev92 commented:
Looks fine now :+1: 

Although I expect the grenadiers will need a damage buff.
Not merging because not sure of @pchote's +1

pchote commented:
My :+1: above was nullified by the complete change (also by my insufficient testing not spotting those major issues).

obrakmann commented:
:+1:

obrakmann commented:
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-%28bleed%29/_compare/b054cd351ea6c67cb1a1472421026233a8a09173)
